books/RayTracingTheNextWeek.html

```diff
@@ -2904,24 +2951,112 @@
   $$ x' =  \cos(\theta) \cdot x + \sin(\theta) \cdot z $$
   $$ z' = -\sin(\theta) \cdot x + \cos(\theta) \cdot z $$
 
-And about the x-axis:
+And if we want to rotate about the x-axis:
 
   $$ y' = \cos(\theta) \cdot y - \sin(\theta) \cdot z $$
   $$ z' = \sin(\theta) \cdot y + \cos(\theta) \cdot z $$
 </div>
 
-Unlike the situation with translations, the surface normal vector also changes, so we need to
-transform directions too if we get a hit. Fortunately for rotations, the same formulas apply. If you
-add scales, things get more complicated. See the web page https://in1weekend.blogspot.com/ for links
-to that.
+For a simple operation like a translation, thinking of the operation as a simple movement of the
+initial ray is a fine way to reason about what's going on. But, for a more complex operation like a
+rotation, it can be easy to accidentally get your terms crossed (or forget a negative sign), so it's
+better to consider a rotation as a change of coordinates. The pseudocode for the `translate::hit`
+function above describes the function in terms of _moving_:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    bool hit(const ray& r, interval ray_t, hit_record& rec) const override {
+        // Move the ray backwards by the offset
+
+        // Determine where (if any) an intersection occurs along the offset ray
+
+        // Move the intersection point forwards by the offset
+    }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [translate-hit-move]: <kbd>[hittable.h]</kbd> The moving translate hit code]
```

Comment:
Let's replace this and the code/pseudo-code below with simple numbered lists.

Reply:
Done